File: powervoice-frontend/src/app/services/i-crud-service.ts
import { Observable } from "rxjs";

export interface ICrudService<T> {

    apiUrl: string;

    get(termoBusca?: string): Observable<T[]>;

    getById(id: number): Observable<T>;

    insert(objeto: T): Observable<T>;

    update(objeto: T): Observable<T>;

    delete(id: number): Observable<void>;


}

File: powervoice/src/main/java/com/example/powervoice/service/ICrudService.java
package com.example.powervoice.service;

import java.util.List;

import org.springframework.data.domain.Page;

public interface ICrudService<T> {
    
    public List<T> getAll();

    public T getById(Long id);

    public Page<List<T>> getByAll(String termoBusca, int page, int size);

    public T save(T objeto);

    public void delete(Long id);

}
